### PR TITLE
Create an EnforceObjectsOneToOne plugin [WIP]

### PR DESCRIPTION
**Pull request by bethac07**
Replaces #238 

We initially looked at putting this functionality into RunCellpose (#227), and a lot of the pseudocode happened there, but in writing this up I realized that a) it was fiddly and b) the more DL segmentation stuff we add, the more we were going to need it, and so I extracted it out into its own module. Initial draft here lacks a display and some documentation but otherwise as far as I can tell is otherwise functional and is creating the correct outputs.

### CODE REVIEW

**Comment by bethac07** [comment]
@sugan89 identified a case where this is failing, will dig in later but I now have the files to do so

**Comment by bethac07** [comment]
image

**Comment by bethac07** [comment]
image

File: active_plugins/enforceobjectsonetoone.py [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+        new_secondary_objects.segmented = secondary_seg
+        if pre_secondary.has_parent_image:
+            new_secondary_objects.parent_image = pre_secondary.parent_image
+        #we are NOT handling ie unedited segmented, since those are rarely made in DL
+
+        workspace.object_set.add_objects(new_primary_objects, self.output_primary_objects_name.value)
+
+        workspace.object_set.add_objects(new_secondary_objects, self.output_secondary_objects_name.value)
+
+        #relate new primary to new secondary, and get the measurements
```
Comment (bethac07) [inline active_plugins/enforceobjectsonetoone.py:178]: TODO - these comments are out of order

**Comment by ErinWeisbart** [comment]
Would like to have the third panel of the display image be outlines of the enforced primary and secondary objects superimposed but I don't understand how to make that work so I need help. 

Do we ever have module inputs that are only for the display? Because it would be neat to add the functionality of `Do you want to see enforced objects superimposed on an image?` and then `Select image` and then having that as the background of the outline image as it would be much easier to see if your new enforced objects make biological sense (without having to use the Workspace Viewer).

**Comment by bethac07** [comment]
>Would like to have the third panel of the display image be outlines of the enforced primary and secondary objects superimposed but I don't understand how to make that work so I need help.

IDSecondary has something like that, perhaps that code will be of use?

>Do we ever have module inputs that are only for the display? Because it would be neat to add the functionality of Do you want to see enforced objects superimposed on an image? and then Select image and then having that as the background of the outline image as it would be much easier to see if your new enforced objects make biological sense (without having to use the Workspace Viewer).

Oh @ErinWeisbart , the answer to "do we ever do X in CellProfiler" is ALWAYS yes. In this case, the "display local maxima" setting in IDPrimary

**Comment by ErinWeisbart** [comment]
I still don't understand how, but I did it :)
AFAIK this is ready to merge (other than @bethac07 's outstanding comment above about comments being out of order)

**Comment by bethac07** [comment]
I confess I found some of the descriptions over-mathy, to the point in one or two places I couldn't parse correctness, so I did a small editing pass. If you can live with @ErinWeisbart , please do pull, otherwise, edit as you see fit and then pull!

**Pull request merged**